operator/pkg/central/values/translation/translation.go

```diff
@@ -70,8 +76,19 @@ func translate(c platform.Central) (chartutil.Values, error) {
 	if centralSpec == nil {
 		centralSpec = &platform.CentralComponentSpec{}
 	}
-
-	v.AddChild("central", getCentralComponentValues(centralSpec))
+	annotations := c.GetAnnotations()
+	var obsoletePvc bool
+	if value, ok := annotations[common.CentralPVCObsoletedAnnotation]; ok {
+		obsoletePvc = strings.EqualFold("true", strings.TrimSpace(value))
+	}
+	checker := &pvcExistenceChecker{
+		ctx:         ctx,
+		client:      t.Client,
+		nameSpace:   c.GetNamespace(),
+		obsoletePvc: obsoletePvc,
```

Comment:
Continuing the comment from `pvcExistenceChecker struct`, I think it would be better to keep the checker focused, and instead change the signature of `getCentralComponentValues` to let it interrogate the annotations on the CR itself (using the new extracted function mentioned above). Either by makit it take a whole `Central` rather than just `CentralSpec`, or by making it accept an annotations map.

Reply:
Any comment on the above two comments about moving the `obsoletePvc` bool out of the checker @c-du ?

Reply:
yes, I changed its name from existence checker to pvcStateChecker so that it can include both existence and obsolete state.

Reply:
Part of the problem is that existence checks **state** and the obsolete bit is part of the **spec**. Keeping these two distinct in k8s operators is quite important, so I'd say it's another reason to **not** include it in the checker.